### PR TITLE
Add markdown linting

### PR DESCRIPTION
**Pull request by jsumners**
This PR adds a workflow that will check the line length within Markdown documents in this project. It's possible we could migrate this to the common workflows repo, but I'm not clear if that is possible given how the tools in this workflow work. Anyone is welcome to do their own testing and move it if they determine it viable.

See https://github.com/stupid-org/md-lint/pull/1/files for an example of what a failed check looks like.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/md-lint.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  build:
+    name: Lint Markdown
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@main
+
+      - name: Setup Node
+        uses: actions/setup-node@main
```
Comment (Fdawgs) [inline .github/workflows/md-lint.yml:23]:
```suggestion
      - name: Checkout Code
        uses: actions/checkout@v3
        with:
          persist-credentials: false
        
      - name: Setup Node
        uses: actions/setup-node@v3
```

File: .github/workflows/md-lint.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+on:
+  push:
+    branches-ignore:
+      - master
+      - main
+  pull_request:
+    branches:
+      - master
+      - main
```
Comment (Fdawgs) [inline .github/workflows/md-lint.yml:11]:
```suggestion
  push:
    branches:
      - master
      - main
    paths:
       - "**/*.md"
  pull_request:
    branches:
      - master
      - main
    paths:
       - "**/*.md"
```

Something like this, so it only triggers when someone is changing a markdown file?

**Review by Fdawgs — CHANGES_REQUESTED** [review]

File: .github/workflows/md-lint.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          persist-credentials: false
+        
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 'lts/*'
+
+      - name: Install Linter
+        run: npm i --no-save markdownlint-cli2
```
Comment (mcollina) [inline .github/workflows/md-lint.yml:34]: Add this to devDependencies instead and add it to our scripts too.

File: .github/workflows/md-lint.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+      - name: Add Matcher
+        uses: xt0rted/markdownlint-problem-matcher@b643b0751c371f357690337d4549221347c0e1bc
```
Comment (mcollina) [inline .github/workflows/md-lint.yml:37]: What's this exactly?

**Review by mcollina — CHANGES_REQUESTED** [review]
This setup only runs in CI. I think we should be able to run locally too. I think `remark` might offer a better toolchain but I'm ok with anything I can run locally as well.

(In Node.js TSC we use https://github.com/nodejs/TSC/blob/main/package.json#L7)

File: .github/workflows/md-lint.yml (hunk 4 above)
Comment (jsumners) [reply .github/workflows/md-lint.yml:37]: This provides the inline errors on the `files` tab of a pull request.

File: .github/workflows/md-lint.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          persist-credentials: false
+        
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 'lts/*'
+
+      - name: Install Linter
+        run: npm install
```
Comment (Fdawgs) [inline .github/workflows/md-lint.yml:34]:
```suggestion
        run: npm install --ignore-scripts
```

**Comment by Fdawgs** [comment]
Had to remove it as a Required workflow from the main branch, see https://github.com/fastify/fastify/pull/4005#issuecomment-1153877750

**Comment by jsumners** [comment]
It's a really basic implementation. It should really be implemented as "do markdown changes exist? yes: run check, no: return green".

**Pull request merged**